source/large_message_sub_pub_demo.c

```diff
@@ -197,23 +209,38 @@ void vStartLargeMessageSubscribePublishTask( configSTACK_DEPTH_TYPE uxStackSize,
 
 /*-----------------------------------------------------------*/
 
-static void prvSubscribeCommandCallback( void *pxCommandContext,
+static void prvSubscribeCommandCallback( void * pxCommandContext,
                                          MQTTAgentReturnInfo_t * pxReturnInfo )
 {
-CommandContext_t *pxApplicationDefinedContext = ( CommandContext_t * ) pxCommandContext;
+    CommandContext_t * pxApplicationDefinedContext = ( CommandContext_t * ) pxCommandContext;
 
     /* Store the result in the application defined context so the calling task
-    can check it. */
+     * can check it. */
     pxApplicationDefinedContext->xReturnStatus = pxReturnInfo->returnCode;
+
+    /* Check if the subscribe operation is a success. Only one topic is
+     * subscribed by this demo. */
+    if( ( pxReturnInfo->returnCode == MQTTSuccess ) &&
+        ( pxReturnInfo->pSubackCodes[ 0 ] != MQTTSubAckFailure ) )
```

Comment:
One of these is redundant, since the return code is `MQTTSuccess` if and only if every status code in the array is not 0x80 (`MQTTSubAckFailure`)